src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -362,80 +428,255 @@ private void ToggleGesture(RoutedEvent routedEvent)
+		public IAsyncOperation<DataPackageOperation> StartDragAsync(PointerPoint pointerPoint)
+			=> AsyncOperation.FromTask(ct => StartDragAsyncCore(pointerPoint, _pendingRaisedEvent.args, ct));
+
+		private Task<DataPackageOperation> StartDragAsyncCore(PointerPoint pointer, PointerRoutedEventArgs ptArgs, CancellationToken ct)
 		{
-			// When we bubble a gesture event from a child, we make sure to abort any pending gesture/manipulation on the current element
-			if (routedEvent == HoldingEvent)
+			ptArgs ??= CoreWindow.GetForCurrentThread()!.LastPointerEvent as PointerRoutedEventArgs;
+			if (ptArgs is null || ptArgs.Pointer.PointerDeviceType != pointer.PointerDevice.PointerDeviceType)
 			{
-				if (_gestures.IsValueCreated)
-				{
-					_gestures.Value.PreventHolding(((HoldingRoutedEventArgs) args).PointerId);
-				}
+				// Fairly impossible case ...
+				return Task.FromResult(DataPackageOperation.None);
 			}
-			else
+
+			// Note: originalSource = this => DragStarting is not actually a routed event, the original source is always the sender
+			var routedArgs = new DragStartingEventArgs(this, ptArgs);
+			PrepareShare(routedArgs.Data); // Gives opportunity to the control to fulfill the data
+			SafeRaiseEvent(DragStartingEvent, routedArgs); // The event won't bubble, cf. PrepareManagedDragAndDropEventBubbling
+
+			// TODO: Add support for the starting deferral!
+
+			if (routedArgs.Cancel)
 			{
-				// Note: Here we should prevent only the same gesture ... but actually currently supported gestures
-				// are mutually exclusive, so if a child element detected a gesture, it's safe to prevent all of them.
-				CompleteGesture(); // Make sure to set the flag _isGestureCompleted, so won't try to recognize double tap
+				// The completed event is not raised if the starting has been cancelled
+				return Task.FromCanceled<DataPackageOperation>(CancellationToken.None);
+			}
+
+			if (!pointer.Properties.HasPressedButton)
+			{
+				// This is the UWP behavior: if no button is pressed, then the drag is completed immediately
+				var noneResult = Task.FromResult(DataPackageOperation.None);
+				OnDragCompleted(noneResult, this);
+				return noneResult;
 			}
+
+			var result = new TaskCompletionSource<DataPackageOperation>();
+			result.Task.ContinueWith(OnDragCompleted, this, TaskContinuationOptions.NotOnCanceled | TaskContinuationOptions.RunContinuationsAsynchronously);
+
+			var dragInfo = new CoreDragInfo(
+				source: ptArgs,
+				data: routedArgs.Data.GetView(),
+				routedArgs.AllowedOperations,
+				dragUI: routedArgs.DragUI);
+			dragInfo.RegisterCompletedCallback(result.SetResult);
+
+			CoreDragDropManager.GetForCurrentView()!.DragStarted(dragInfo);
+
+			return result.Task;
+		}
+
+		private static void OnDragCompleted(Task<DataPackageOperation> resultTask, object snd)
+		{
+			var that = (UIElement)snd;
+			var result = resultTask.IsFaulted
+				? DataPackageOperation.None
+				: resultTask.Result;
```

Comment:
Codacy found an issue: [Replace this use of 'Task.Result' with 'await'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6333952)